### ISSUE TITLE
New Striatal Scaffold Model

### ISSUE THREAD

**Issue by Hjorthmedh** [cnr-ibf-pa/hbp-bsp-issues#378]
## New Use Case
  
Aspect | Detail
------------ | -------------
Summary | Updated version of the Striatum Scaffold model. New touch detection, improved cell models, etc.
Usecase Group | Microcircuitry
Expert | Johannes Hjorth
Scientific User |
Deadline | Christmas 2019?
Target audience | Developers
Target interface | Notebook
HPC Requirements | Possibly.
Dependencies |
Nominal runtime |
  
## List of features
Create striatal microcircuitry: Place neurons (mandatory), touch detection (mandatory), prune synapses (mandatory). Run simulation (important). Visualise (nice to have)
 
    
## Acceptance Criteria
Runnable in Notebook.
Passing validation against experimental data.
Documented.
  
### System
Parallell version will require HPC access if larger networks should be run.

### Performance
Full striatum takes approximately 3 days on 12 core workstation for microcircuitry creation.

**Comment by Hjorthmedh** [comment]
Working on getting correct connectivity for ChIN neurons.

**Comment by Hjorthmedh** [comment]
ChIN connectivity almost finished tuned.
Next step LTS connectivity.

**Comment by Hjorthmedh** [comment]
120,000 neuron microcircuit created using the PDC's cluster Tegner. The job consists of parallel and serial parts. I want to split these into different jobs, to avoid idle workers during the merging of data files which is done in serial. Will then try and get the code to run on the HBP super computers.

We have preliminary synaptic connectivities for dSPN, iSPN, FS, LTS and ChIN now. Currently starting to look at synaptic dynamics and specifying input to be used in the simulations.

**Comment by Hjorthmedh** [comment]
Working on synaptic dynamics for cortical and thalamic input to striatum.

We are now also writing on a manuscript.

**Comment by Hjorthmedh** [comment]
Optimised a first version of the dynamical parameters for the synapses. Working on a second iteration - need to set the scale of the conductance properly.

Updated the code base so that the network can support different dynamics for different synapses. There is now a parameterID for each synapse, which picks a model from a family of synapse models.

We will revisit the external synaptic input to the model, in particular frequency and correlation.

**Comment by Hjorthmedh** [comment]
Finished second iteration of optimisations, resulting in much better fits. 

This week starting a revision of the external input for the network model.

**Comment by Hjorthmedh** [comment]
Have done a few initial striatum network simulations with the updated network. Working on scaling it up and tuning the input frequencies of cortical and thalamic input for the different neuron types.

**Comment by Hjorthmedh** [comment]
We have run 2000 neuron network on beskow for SfN poster. Updated the gap junction code.

**Comment by Hjorthmedh** [comment]
We are working on migrating the code base to github from kth's github. Also preparing figures for article.

**Comment by Hjorthmedh** [comment]
Codebase is on Github, private repository.

Tuned intra-striatal synapses between dSPN, iSPN and FS neurons to surrogate data generated from experimental parameters.

Validated conductance for synapses between dSPN, iSPN and FS by triggering a spike in neurons one at a time, and checking the response in the connected neighbours and compared results with experimental data.

Updated connectivity pruning: f1 and softMax function, to reduce the long tail for number of synapses between a connected pair.

In the process of writing a validation for the connectivity using data from Chuhuma 2011 where 10% of SPN are activated and response in dSPN, ChIN and FS are recorded.

**Comment by Hjorthmedh** [comment]
Written virtual experiments to match Planert et al 2010, Straub et al 2016, Chuhma et al 2011.

Fixed a crash bug in merge of synapses files that only manifests for certain number of workers. 

Fixed a bug that crashed input generation when running in parallel.

Modified LTS connectivity. 

Currently generating new figures for article with the new virtual experiments etc.

**Comment by Hjorthmedh** [comment]
Have created a tutorial example of Snudda for the Pavia hackathon, will adapt it into a use case.

Article submitted last week. Preparing code for release.

**Comment by Hjorthmedh** [comment]
https://collab.humanbrainproject.eu/#/collab/9102/nav/520440

New striatum network model use case.

Waiting on a reset of my environment before I can proceed. Also need Neuron 7.7 or later installed.

**Comment by Hjorthmedh** [comment]
Alex, the Neuron version installed right now seems to be for Python 2, I get error messages about print not having parenthesis. So we need Neuron 7.7 (or later) installed for Python 3 on the collab.

**Comment by Hjorthmedh** [comment]
Have now got access to lab.humanbrainproject.eu for the new environment. Uploaded the old python notebook file, and now looking for a way to load it. Then should just need to update the load path from which it gets the file, and that should hopefully just be running after that.

**Comment by alex4200** [comment]
@Hjorthmedh  Hi, what is the status on this task? Should it be moved to SGA3?

**Comment by Hjorthmedh** [comment]
Yes please move it to SGA3, the notebook runs locally, but there was a problem with the image that is used on HBP labs platform. They updated once with a fix, but that fix did not help. Have been busy and not had time to check if there was a second fix.
